nodejs/week2/app.js

```diff
@@ -0,0 +1,103 @@
+import express from 'express';
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+// Support parsing JSON requests
+app.use(express.json());
+
+const getDocuments = () => {
+    try {
+        const filePath = path.join(__dirname, 'documents.json');
+        const data = fs.readFileSync(filePath);
+        return JSON.parse(data);
+    } catch (error) {
+        console.error('Error reading documents file:', error);
+        return [];
+    }
+};
+
+app.get("/", (req, res) => {
+    res.send("This is a search engine");
+});
+
+// GET /search endpoint
+app.get("/search", (req, res) => {
+    const query = req.query.q;
+    const documents = getDocuments();
+    if (!query) {
+        return res.json([]);
+    }
+    const filteredDocuments = documents.filter(doc =>
+        doc.value && doc.value.includes(query)
```

Comment:
Your filter is [case-sensitive](https://www.lifewire.com/what-does-case-sensitive-mean-2625824). Would be nice to make case-insensitive